src/test/FuzzerImpl.cpp

```diff
@@ -1600,6 +1544,66 @@ TransactionFuzzer::reduceNativeBalancesAfterSetup(AbstractLedgerTxn& ltxOuter)
     ltx.commit();
 }
 
+void
+TransactionFuzzer::reduceTrustLineBalancesAndLimitsAfterSetup(
+    AbstractLedgerTxn& ltxOuter)
+{
+    LedgerTxn ltx(ltxOuter);
+
+    xdr::xvector<Operation> ops;
+
+    // Reduce trustline balances and limits so that fuzzing has a better
+    // chance of exercising edge cases.
+    for (auto const& trustLine : trustLineParameters)
+    {
+        auto const trustor = trustLine.mTrustor;
+        PublicKey account;
+        FuzzUtils::setShortKey(account, trustor);
+
+        auto const asset = trustLine.mAssetID.toAsset();
+
+        PublicKey issuer;
+        FuzzUtils::setShortKey(issuer, trustLine.mAssetID.mIssuer);
+
+        // Reduce "account"'s balance of this asset by paying the
+        // issuer.
+        auto tle = stellar::loadTrustLine(ltx, account, asset);
+        auto const availableTLBalance =
+            tle.getAvailableBalance(ltx.loadHeader());
+        auto const targetAvailableTLBalance =
+            trustLine.mAssetAvailableForTestActivity;
+        auto const paymentAmount =
+            availableTLBalance - targetAvailableTLBalance;
+
+        assert(availableTLBalance > targetAvailableTLBalance);
+        auto reduceNonNativeBalanceOp =
+            txtest::payment(issuer, asset, paymentAmount);
+        reduceNonNativeBalanceOp.sourceAccount.activate() =
+            toMuxedAccount(account);
+        ops.emplace_back(reduceNonNativeBalanceOp);
+
+        // Reduce this trustline's limit.
```

Comment:
I've been thinking about this for a few commits now, but it makes even more sense after this refactoring. I think this would be simpler if you first applied the balance updates, then used the new state to compute the limit updates. Then you can use existing functions to avoid reinventing the wheel.

Reply:
Good idea, that does make it nicer.  Done.